pkg/kubelet/kubelet_pods.go

```diff
@@ -487,6 +502,16 @@ func managedHostsFileContent(hostIPs []string, hostName, hostDomainName string,
 	return buffer.Bytes()
 }
 
+// isSupportIPv6 determines whether the host supports ipv6 by check the global ipv6 support sysctl.
+func isSupportIPv6() bool {
+	sysctl := utilsysctl.New()
+	if val, err := sysctl.GetSysctl(sysctlAllDsiableIPv6); err == nil && val == 0 {
```

Comment:
I verified that this would cover us on our systems where we disable via the kernel commandline too. When disabled via the command line, or not compiled in, this sysctl is not present, so since you treat the error as false, this works well.

My only question is why not go with the solution of checking the IPs assigned to the loopback interface?  Since that would cover case were ipv6 is enabled on the system but hasn't been configured on the loopback interface by CNI

Reply:
We have done some discussion here before. Kubelet doesn't know about the loopback interface of the pod, so we use the method of verifying sysctl.
ref: https://github.com/kubernetes/kubernetes/pull/124758#discussion_r1595693979 https://github.com/kubernetes/kubernetes/pull/124758#discussion_r1596033950